### ISSUE TITLE
[Android] TalkBack includes CollectionView Header and EmptyView in list count

### ISSUE THREAD

**Issue by wboekhout** [dotnet/maui#35681]
### Description

The CollectionView has the option to add a header and footer to it, as well as defining an empty view. The problem is that these views are made part of the list that is generated on Android. This makes it that as soon as you focus on an item that is part of the CollectionView, that TalkBack will pronounce the wrong number of items inside the list. Or it will make the wrong announcement that you are already inside the list.

On Android, the CollectionView becomes a `RecyclerView`, which is some kind of list view that contains the header and footer as list items, alongside the items based on `CollectionView.ItemsSource`. There is no distinction being made between them.

This makes it so that TalkBack does not work properly when using `CollectionView.Header` and `CollectionView.Footer`. I would like to use them so that the header and footer scroll with the rest of the items. Having a separate header and footer means that my scrollable area of my page is reduced, which is a problem if my app needs to be used in landscape mode.

<img width="398" height="865" alt="Image" src="https://github.com/user-attachments/assets/c7885960-b4ab-4ebf-bf1d-a556b0b6b9c3" />

https://github.com/user-attachments/assets/a4fc7809-577f-4b0e-9312-ce2057f4ffa4

### Steps to Reproduce

1. Use the MAUI solution from the linked repository.
2. Run the app and use TalkBack to navigate over the elements.

When focused on the header, TalkBack will read "in list, 3 items". It counts the header, EmptyView, and footer as three items. What should happen is that none of these items should be considered as part of the list.

If an `ItemsSource` with 3 items is included, then TalkBack will read "in list, 5 items" when initially focusing on the CollectionView. So it will count the header + footer + 3 list items.

### Link to public reproduction project repository

https://github.com/wboekhout/Maui-CollectionView-TalkBack-Issue

### Version with bug

10.0.70

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 16

### Did you find any workaround?

Not yet.

### Relevant log output

```shell

```

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/maui/issues/2674

**Comment by LogishaSelvarajSF4525** [comment]
This issue has been validated in Visual Studio 1.122.1 with .NET MAUI versions 9.0.120, 10.0.60 & 10.0.70. It was reproducible on Android platform. I have attached the video for reference:

Issue reproduced:

https://github.com/user-attachments/assets/2ab939ba-bbad-4e09-938a-d42df7071995
